src/NUnitFramework/tests/Constraints/IndexerConstraintTests.cs

```diff
@@ -0,0 +1,177 @@
+            Assert.That(tester, Has.Item(41).EqualTo("Still calculating").And.Item(42).EqualTo("Answer to the Ultimate Question of Life, the Universe, and Everything"));
+
+            Assert.That(tester, Has.Item(string.Empty).EqualTo("Second indexer"));
+            Assert.That(tester, Has.Item(1, 2).EqualTo("Third indexer"));
+        }
+
+        [Test]
+        public void DoesNotMatchMissingIndexer()
+        {
+            var expectedErrorMessage = $"  Expected: Default indexer accepting arguments < 42.0d >{NL}  But was:  <NUnit.Framework.Constraints.IndexerConstraintTests+IndexerTester>{NL}";
+            
+            var tester = new IndexerTester();
+            
+            var ex = Assert.Throws<AssertionException>(() => Assert.That(tester, Has.Item(42d)));
+            Assert.That(ex.Message, Is.EqualTo(expectedErrorMessage));
+        }
+
+        [Test]
+        public void DoesNotMatchMissingIndexerEquality()
+        {
+            var expectedErrorMessage = $"  Expected string length 14 but was 13. Strings differ at index 0.{NL}  Expected: \"Second indexer\"{NL}  But was:  \"Third indexer\"{NL}  -----------^{NL}";
+            
+            var tester = new IndexerTester();
+
+            var ex = Assert.Throws<AssertionException>(() => Assert.That(tester, Has.Item(4, 2).EqualTo("Second indexer")));
+            Assert.That(ex.Message, Is.EqualTo(expectedErrorMessage));
+        }
+
+        [Test]
+        public void CanMatchGenericIndexer()
+        {
+            var tester = new GenericIndexerTester<int>(100);
+
+            Assert.That(tester, Has.Item(42));
+            Assert.That(tester, Has.Item(42).EqualTo(100));
+        }
+
+        [Test]
+        public void CanMatchNamedIndexer()
+        {
+            var tester = new NamedIndexTester();
+
+            Assert.That(tester, Has.Item(42));
+            Assert.That(tester, Has.Item(42).EqualTo("A Named Int Indexer"));
+            Assert.That(tester, Has.Item(42, 43).EqualTo("A Named Int Int Indexer"));
+        }
+
+        [Test]
+        public void CanMatchShadowIndexer()
+        {
+            var tester = new DerivedClassWithoutIndexer();
+
+            Assert.That(tester, Has.Item(42));
+            Assert.That(tester, Has.Item(string.Empty).EqualTo("Second indexer from shadow"));
+            Assert.That(tester, Has.Item(1, 2).EqualTo("Third indexer from shadow"));
+            Assert.That(tester, Has.Item(41).EqualTo("Still calculating"));
+        }
+
+        [Test]
+        public void CanMatchShadowNamedIndexer()
+        {
+            var tester = new DerivedClassWithoutNamedIndexer();
+
+            Assert.That(tester, Has.Item(42));
+            Assert.That(tester, Has.Item(42).EqualTo("A Named Int Indexer from shadow"));
+            Assert.That(tester, Has.Item(42, 43).EqualTo("A Named Int Int Indexer"));
+        }
+
+        private class IndexerTester
+        {
+            public int DummyParam { get; set; }
```

Comment:
While playing with the reflection code, those where to ensure it was searching for indexers only and not picking up other fields. Since then the search has been improved as not to be "implement" a coded search pattern. Hence, we could remove these if you would like?

Reply:
Got that!
No, it makes sense, I suppose.